### PR TITLE
Fix log issue with @QuarkusIntegrationTest

### LINKED ISSUE #33229: Native binary run log is not visible when running tests with Quarkus main
### Describe the bug

Native binary run log is not visible when running tests with Quarkus main.

I can see this warning in log -`[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /Users/rsvoboda/Downloads/code-with-quarkus/target/failsafe-reports/2023-05-09T17-02-55_091-jvmRun1.dumpstream`

Quarkus 3.0.2.Final shows the output correctly for native binary.

999-SNAPSHOT
```
[INFO] Running org.acme.GreetingResourceIT
Executing "/Users/rsvoboda/Downloads/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/rsvoboda/Downloads/code-with-quarkus/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.platform.version=999-SNAPSHOT -Dquarkus.platform.group-id=io.quarkus"
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.857 s - in org.acme.GreetingResourceIT
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /Users/rsvoboda/Downloads/code-with-quarkus/target/failsafe-reports/2023-05-09T17-02-55_091-jvmRun1.dumpstream
```

3.0.2.Final
```
[INFO] Running org.acme.GreetingResourceIT
Executing "/Users/rsvoboda/Downloads/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/rsvoboda/Downloads/code-with-quarkus/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO"
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2023-05-09 16:58:56,819 INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT native (powered by Quarkus 3.0.2.Final) started in 0.025s. Listening on: http://0.0.0.0:8081
2023-05-09 16:58:56,820 INFO  [io.quarkus] (main) Profile prod activated.
2023-05-09 16:58:56,820 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.1 s - in org.acme.GreetingResourceIT
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

### Expected behavior

Quarkus main shows the output correctly for native binary.

### Actual behavior

Quarkus main doesn't shows the output correctly for native binary. 

this warning is in console: `[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /Users/rsvoboda/Downloads/code-with-quarkus/target/failsafe-reports/2023-05-09T17-02-55_091-jvmRun1.dumpstream`

### How to Reproduce?

Get sample app from https://code.quarkus.io/

Run `mvn clean verify -Dnative -Dquarkus.platform.group-id=io.quarkus -Dquarkus.platform.version=999-SNAPSHOT` 

Or just `mvn clean verify` to see Quarkus 3.0.2.Final (correct) behavior.

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

Java 17

### GraalVM version (if different from Java)

22.3

### Quarkus version or git rev

Quarkus main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by geoand**
Fixes: #33229

### CODE REVIEW

**Comment by geoand** [comment]
This will need to be in Quarkus `3.1` but **not** `3.0` (as that stream does not contain the problematic commit)

**Comment by gsmet** [comment]
@geoand yup, I just branched and the standard backport label is now for 3.1

**Comment by geoand** [comment]
💪🏼

File: test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java [hunk 1]
```diff
@@ -307,6 +312,31 @@ private void unableToDetermineData(String errorMessage) {
...
+            this.inputStream = inputStream;
+        }
+
+        @Override
+        public void run() {
+            byte[] b = new byte[100];
+            int i;
+            try {
+                while ((i = inputStream.read(b)) > 0) {
+                    System.out.print(new String(b, 0, i, StandardCharsets.UTF_8));
```
Comment (machi1990) [inline test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java:332]: Do you think it is worth to make the printing configurable? `System.out.print vs System.err.print` depending on the input system in https://github.com/quarkusio/quarkus/blob/6acbce514fff84ff30792714d45e7e02e0b5422f/test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java#L60-L61?

Comment (geoand) [reply test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java:332]: No one has ever asked for it and it's not been a problem so I prefer to leave it as is

File: test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java [hunk 2]
```diff
@@ -49,12 +51,15 @@ public static Config installAndGetSomeConfig() {
 
     /**
      * Launches a process using the supplied arguments and makes sure the process's output is drained to standard out
+     * <p>
+     * Implementation detail: Avoid using ProcessBuilder's redirect here because it causes problems with Maven Failsafe
+     * as can be seen in <a href="https://github.com/quarkusio/quarkus/issues/33229">...</a>
```
Comment (gastaldi) [inline test-framework/common/src/main/java/io/quarkus/test/common/LauncherUtil.java:56]:
```suggestion
     * as can be seen in <a href="https://github.com/quarkusio/quarkus/issues/33229">here</a>
```

**Review by rsvoboda — APPROVED** [review]
I have tried the patch and the log is now visible

```
[INFO] Running org.acme.GreetingResourceIT
Executing "/Users/rsvoboda/Downloads/code-with-quarkus/target/code-with-quarkus-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/rsvoboda/Downloads/code-with-quarkus/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO"
__  ____  __  _____   ___  __ ____  ______
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/
2023-05-17 10:13:09,277 INFO  [io.quarkus] (main) code-with-quarkus 1.0.0-SNAPSHOT native (powered by Quarkus 999-SNAPSHOT) started in 0.028s. Listening on: http://0.0.0.0:8081
2023-05-17 10:13:09,278 INFO  [io.quarkus] (main) Profile prod activated.
2023-05-17 10:13:09,278 INFO  [io.quarkus] (main) Installed features: [cdi, resteasy-reactive, smallrye-context-propagation, vertx]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 3.068 s - in org.acme.GreetingResourceIT
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 1, Failures: 0, Errors: 0, Skipped: 0
```

**Pull request merged**